core/src/main/java/com/linecorp/armeria/client/AbstractEventLoopState.java

```diff
@@ -29,9 +29,9 @@ static AbstractEventLoopState of(List<EventLoop> eventLoops, int maxNumEventLoop
                                      DefaultEventLoopScheduler scheduler) {
         if (maxNumEventLoops == 1) {
             return new OneEventLoopState(eventLoops, scheduler);
+        } else if (maxNumEventLoops <= 128) {
+            return new ArrayBasedEventLoopState(eventLoops, maxNumEventLoops, scheduler);
         }
-        // TODO(minwoox) Introduce array based state which is used when the maxNumEventLoops is greater than 1
-        //               and less than N for the performance.
         return new HeapBasedEventLoopState(eventLoops, maxNumEventLoops, scheduler);
```

Comment:
For readability and code aesthetics, we may choose early return or if else chain.
```java
if {
    ...
} else if {
    ...
} else {
   ...
}

// or 
if (...) {
    return
}

if (...) {
    return
}

return ...
```

Reply:
+1 for the early return. 😉